### PR TITLE
Add type narrow test for `if-else-stmt`, `while-stmt` and `ternary-expr`

### PR DESCRIPTION
**Pull request by lochana-chathura**
## Purpose
$subject. Resolves a part of #34631

## Approach
N/A

## Samples
N/A

## Remarks
N/A

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/expressions/ternary/ternary_expr_type_narrow.bal [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        _ = x is int ? intFn(x) : booleanFn(x); // OK
+    }
+
+    int|string|boolean|decimal y = "s";
+    if y is string {
+
+    } else {
+        while y !is decimal {
+            _ = y is int ? intFn(y) : booleanFn(y); // OK
+
```
Comment (pcnfernando) [inline tests/jballerina-unit-test/src/test/resources/test-src/expressions/ternary/ternary_expr_type_narrow.bal:61]:
```suggestion
```

File: tests/jballerina-unit-test/src/test/resources/test-src/expressions/ternary/ternary_expr_type_narrow.bal [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        }
+    }
+
+    while y !is string {
+        if !(y is decimal) {
+            _ = y is int ? intFn(y) : booleanFn(y); // OK
+
```
Comment (pcnfernando) [inline tests/jballerina-unit-test/src/test/resources/test-src/expressions/ternary/ternary_expr_type_narrow.bal:68]:
```suggestion
```

File: tests/jballerina-unit-test/src/test/resources/test-src/expressions/ternary/ternary_expr_type_narrow.bal [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        }
+    }
+
+    while y !is string {
+        while y !is decimal {
+            _ = y is int ? intFn(y) : booleanFn(y); // OK
+
```
Comment (pcnfernando) [inline tests/jballerina-unit-test/src/test/resources/test-src/expressions/ternary/ternary_expr_type_narrow.bal:75]:
```suggestion
```

File: tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/test_type_guard_type_narrow_1.bal [hunk 4]
```diff
@@ -0,0 +1,557 @@
...
+        false _ = b;
+        return;
+    } else {
+        () _ = b; // type not narrowed. issue #30598, #33217
+    }
+
+    () _ = b; // type not narrowed. issue #34307
+}
+
+function test10(true|false? b) {
```
Comment (pcnfernando) [inline tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/test_type_guard_type_narrow_1.bal:118]: Let's add tests using !is and logical exp as well if not already covered
```ballerina
function foo(int|string r) {
    if r is int || r !is string {
        int _ = r;  //error incompatible types: expected 'int', found '(int|string)'
    } 
}
```

File: tests/jballerina-unit-test/src/test/resources/test-src/expressions/ternary/ternary_expr_type_narrow.bal [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    while y !is string {
+        while y !is decimal {
+            _ = y is int ? intFn(y) : booleanFn(y); // OK
+        }
+    }
+}
+
+type A record {|
```
[the remark this answers is not in the document]
Comment (lochana-chathura) [reply tests/jballerina-unit-test/src/test/resources/test-src/expressions/ternary/ternary_expr_type_narrow.bal:98]: Added with [5d60bed](https://github.com/ballerina-platform/ballerina-lang/pull/34964/commits/5d60bed97fbba11ae41e45effc929d7bb210a205)

File: tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/test_type_guard_type_narrow_1.bal (hunk 4 above)
Comment (lochana-chathura) [reply tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/test_type_guard_type_narrow_1.bal:118]: We had these covered in `test_type_guard_type_narrow_3.bal`. Anyway, I had written them using the `!(r is string)` form. Duplicated those tests and included `r !is string` form also.

File: tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/test_type_guard_type_narrow_1.bal [hunk 6]
```diff
@@ -0,0 +1,557 @@
...
+function test5(int|string|boolean b) {
+    if b is string {
+        string _ = b;
+        return;
+    } else if b is boolean {
+        boolean _ = b;
+        return;
+    }
+
+    int _ = b; // type not narrowed. issue #34307
```
[the remark this answers is not in the document]
Comment (lochana-chathura) [reply tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/test_type_guard_type_narrow_1.bal:69]: Added. thanks.

**Review by pcnfernando — DISMISSED** [review]

**Comment by pcnfernando** [comment]
Let's add few positive tests as well

**Review by pcnfernando — APPROVED** [review]

**Pull request merged**